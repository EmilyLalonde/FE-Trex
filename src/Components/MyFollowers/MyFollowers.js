import React from 'react';
import { 
  StyleSheet, 
  Text, 
  View, 
  ScrollView,
  TouchableOpacity 
} from 'react-native';
import WandererFooter from '../WandererFooter/WandererFooter';
import WandererHeader from '../WandererHeader/WandererHeader';

export function MyFollowers(props) {

    const {navigate} = props.navigation;


    return (
      <View style={styles.container}>
        <WandererHeader />
        <ScrollView>
          <Text style={styles.title}>My Followers</Text>
          {/* <View>
            {this.followerElements.length > 0 && this.followerElements}
          </View> */}
          <TouchableOpacity style={styles.addFollowerButton}>
            <Text style={styles.text} onPress={() => navigate('FollowerForm')}>Add a New Follower</Text>
          </TouchableOpacity>
        </ScrollView>
        <WandererFooter navigate={navigate} />
    </View>
    )
}

const styles = StyleSheet.create({
  title: {
    fontSize: 30
  }, 
  container: {
    flex: 1,
    backgroundColor: '#000000',
    alignItems: 'stretch',
    justifyContent: 'flex-start'
    
  }, 
  text: {
    color: 'white',
    textAlign: 'center',
    paddingVertical: 10,
    fontSize: 20
  },
  title: {
    textAlign: 'center',
    fontSize: 30,
    color: 'white',
    paddingVertical: 10
  },
  addFollowerButton: {
    backgroundColor: '#1C4263',
    borderWidth: 1,
    borderColor: 'white', 
    color: 'white',
    borderRadius: 8,
    backgroundColor: '#1C4263'
  }
})
