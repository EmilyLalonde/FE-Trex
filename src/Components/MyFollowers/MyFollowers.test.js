import React from "react";
import "react-native";
import {MyFollowers} from "./MyFollowers";
import { shallow } from "enzyme";

describe("MyFollowers", () => {
  it("should return true", () => {
    expect(true).toEqual(true);
  });
  it("should match the snapshot when loading", () => {
    let wrapper = shallow(<MyFollowers navigation={{ navigate: jest.fn() }} />);

    expect(wrapper).toMatchSnapshot();
  });
});
