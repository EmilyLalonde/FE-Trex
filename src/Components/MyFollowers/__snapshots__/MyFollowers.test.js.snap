// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFollowers should match the snapshot when loading 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#000000",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <WandererHeader />
  <ScrollViewMock>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      My Followers
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "backgroundColor": "#1C4263",
          "borderColor": "white",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "white",
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
      >
        Add a New Follower
      </Text>
    </TouchableOpacity>
  </ScrollViewMock>
  <WandererFooter
    navigate={[MockFunction]}
  />
</View>
`;
